server/controllers/ticket/close.php

```diff
@@ -30,15 +30,34 @@ class CloseController extends Controller {
     private $ticket;
 
     public function validations() {
-        return [
-            'permission' => 'user',
-            'requestData' => [
-                'ticketNumber' => [
-                    'validation' => DataValidator::validTicketNumber(),
-                    'error' => ERRORS::INVALID_TICKET
+        $session = Session::getInstance();
+
+        if (Controller::isUserSystemEnabled() || Controller::isStaffLogged()) {
+            return [
+                'permission' => 'user',
+                'requestData' => [
+                    'ticketNumber' => [
+                        'validation' => DataValidator::validTicketNumber(),
+                        'error' => ERRORS::INVALID_TICKET
+                    ]
                 ]
-            ]
-        ];
+            ];
+        } else {
+            return [
+                'permission' => 'any',
+                'requestData' => [
+                    'ticketNumber' => [
+                        'validation' => DataValidator::equals($session->getTicketNumber()),
+                        'error' => ERRORS::INVALID_TICKET
+                    ],
+                    'csrf_token' => [
+                        'validation' => DataValidator::equals($session->getToken()),
+                        'error' => Controller::request('csrf_token') . ' != ' . $session->getToken()
```

Comment:
Oh, I see. I didn't realize. This needs to be fixed in
https://github.com/opensupports/opensupports/blob/master/server/controllers/ticket/comment.php#L66
https://github.com/opensupports/opensupports/blob/master/server/controllers/ticket/get.php#L55